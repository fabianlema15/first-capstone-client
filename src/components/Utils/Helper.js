const Helper = {
  getProductArray(){
    return [
      {element: 'input', type: 'text', id: 'name', label:'Name'},
      {element: 'file', type: 'file', id: 'picture', label:'Picture'},
      {element: 'input', type: 'number', id: 'stock', label:'Stock'},
      {element: 'input', type: 'number', id: 'price', label:'Price'},
      {element: 'textarea', id: 'description', label:'Decription'},
    ]
  },
  getClientsArray(){
    return [
      {element: 'input', type: 'text', id: 'full_name', label:'Full Name'},
      {element: 'input', type: 'text', id: 'address', label:'Address'},
      {element: 'input', type: 'tel', id: 'phone', label:'Phone'},
      {element: 'input', type: 'email', id: 'email', label:'Email'},
    ]
  },
  getUsersArray(options){
    return [
      {element: 'input', type: 'text', id: 'first_name', label:'First Name'},
      {element: 'input', type: 'text', id: 'last_name', label:'Last Name'},
      {element: 'input', type: 'text', id: 'address', label:'Address'},
      {element: 'input', type: 'tel', id: 'phone', label:'Phone'},
      {element: 'input', type: 'email', id: 'email', label:'Email'},
      {element: 'select', id: 'role', label:'Role', options:options},
    ]
  },
  getPromotionsArray(){
    return [
      {element: 'input', type: 'text', id: 'name', label:'Name'},
      {element: 'file', type: 'file', id: 'picture', label:'Picture'},
      {element: 'input', type: 'number', id: 'stock', label:'Stock'},
      {element: 'input', type: 'number', id: 'price', label:'Price'},
      {element: 'textarea', id: 'description', label:'Description'},
    ]
  },
  getPromotionProductArray(options){
    return [
      {element: 'select', id: 'product_id', label:'Product', options:options},
      {element: 'input', type: 'number', id: 'quantity', label:'Quantity'},
    ]
  },
  getOrdersArray(options){
    return [
      {element: 'select', id: 'client_id', label:'Client', options:options},
      {element: 'input', type: 'number', id: 'subtotal', label:'Subtotal'},
      {element: 'input', type: 'number', id: 'tax', label:'Tax'},
      {element: 'input', type: 'number', id: 'total', label:'Total'},
      {element: 'textarea', id: 'observation', label:'Observation'},
    ]
  },
  getOrderPromotionArray(options){
    return [
      {element: 'select', id: 'promotion_id', label:'Promotion', options:options},
      {element: 'textarea', id: 'observation', label:'Observation'},
      {element: 'input', type: 'number', id: 'quantity', label:'Quantity'},
      {element: 'input', type: 'number', id: 'price', label:'Price'},
    ]
  },
  getOrderProductArray(options){
    return [
      {element: 'select', id: 'product_id', label:'Product', options:options},
      {element: 'textarea', id: 'observation', label:'Observation'},
      {element: 'input', type: 'number', id: 'quantity', label:'Quantity'},
      {element: 'input', type: 'number', id: 'price', label:'Price'},
    ]
  },
  serializeObj(obj){
    return obj.reduce((acc, item)=>{
      acc[item.id] = item;
      return acc;
    }, {});
  },
  roleHasAccess(role, path){
    const data = {
      admin: ['menu', 'password', 'report', 'product', 'client', 'user', 'promotion', 'order'],
      manager: ['menu', 'password', 'report', 'product', 'client', 'promotion', 'order'],
      seller: ['menu', 'password', 'client', 'order'],
    }
    let response = false;
    data[role.toLowerCase()].forEach(subPath =>
      {
        if(path.toLowerCase().includes(subPath)) response = true;
      })
    return response;
  },
  objHasEmpty(obj){
    for (const key in obj){
      if (obj[key] === '') return true;
    }
    return false
  }
}

export default Helper;
