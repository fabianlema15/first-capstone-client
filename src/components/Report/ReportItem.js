import React from 'react';
import { ColoredLine } from '../Utils/FormElements'

class ReportItem extends React.Component{

  getItemContent = () => {
    return Object.keys(this.props.objLabel).map((key, idx) => <div key={idx} className="item-description">
          <div><span>{this.props.objLabel[key].label}:</span></div> <div>{this.props.order[key]}</div>
       </div>)
  }

  render(){
    return <div>
      <div>
        {this.getItemContent()}
      </div>
      <ColoredLine color="gray" />
    </div>
  }
}

export default ReportItem;
