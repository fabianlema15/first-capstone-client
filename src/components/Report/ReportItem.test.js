import React from 'react';
import ReactDOM from 'react-dom';
import ReportItem from './ReportItem'
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json'
import Helper from '../Utils/Helper'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
configure({ adapter: new Adapter() });


describe('<ReportItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ReportItem obj={{}} objLabel={{}}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders ReportItem as expected', () => {
	    const tree = renderer.create(<ReportItem obj={{}} objLabel={{}}/>).toJSON();
	    expect(tree).toMatchSnapshot();
	  });

  it('With Context', () => {
    const context = { objName: 'Test', getObjArray: Helper.getProductArray()};
	  const wrapper = shallow(<ReportItem obj={{}} objLabel={{}}/>, { context });
	  expect(toJson(wrapper)).toMatchSnapshot();
	})

})
