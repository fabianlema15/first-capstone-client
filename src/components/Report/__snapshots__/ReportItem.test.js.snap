// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportItem /> Renders ReportItem as expected 1`] = `
<div
  className="item-list"
>
  <div />
  <hr
    style={
      Object {
        "backgroundColor": "gray",
        "color": "gray",
        "height": 3,
      }
    }
  />
</div>
`;

exports[`<ReportItem /> With Context 1`] = `
<div
  className="item-list"
>
  <div />
  <ColoredLine
    color="gray"
  />
</div>
`;
